core/dbt/task/run.py

```diff
@@ -454,10 +454,10 @@ def print_results_line(self, results, execution_time) -> None:
 
     def before_run(self, adapter, selected_uids: AbstractSet[str]) -> None:
         with adapter.connection_named("master"):
+            self.defer_to_manifest()
```

Comment:
The purpose of this method is now to add `defer_relation` to all nodes in the project.

Previously, we needed to know at this point which objects exist in the target database locations (which requires the cache). Now that's cache check is happening within `RuntimeRefResolver`, meaning that we can consistently call `defer_to_manifest` _before_ `populate_adapter_cache` in every task.